packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js

```diff
@@ -21,6 +21,8 @@ module.exports = {
   'eol-last': 'warn',
   // Disallow space between function identifier and application
   'func-call-spacing': 'warn',
+  // Require function names to match the name of the variable or property to which they are assigned
+  'func-name-matching': 'warn',
```

Comment:
So it doesn't warn about shadowing, but it does complain about `func-style` since we have it set to declaration only. 

I suppose it couldn't hurt to have this rule as well, there may be some places that it covers where `func-style` doesn't.